pepper-apis/src/main/java/org/broadinstitute/ddp/elastic/participantslookup/ESParticipantsLookupService.java

```diff
@@ -0,0 +1,148 @@
+
+        // if any extra-proxies found then fetch their data from index "users", add to proxiesResults, and merge results again
+        if (governedUserToProxyExtraSearch.size() > 0) {
+            esResultRows = reMergeResultsWithExtraFoundProxies(
+                    query, resultsMaxCount, governedUserToProxy, governedUserToProxyExtraSearch,
+                    usersEsIndex, proxiesResults, participantsStructuredResults);
+
+        }
+
+        participantsLookupResult.setResultRows(esResultRows);
+    }
+
+    /**
+     * Adds to results of "participants_structured" index the 'proxies' data results (taken from
+     * "users" index results).
+     */
+    private List<ParticipantsLookupResultRow> mergeResults(
+            Map<String, ESUsersIndexResultRow> usersResults,
+            Map<String, ESParticipantsStructuredIndexResultRow> participantsStructuredResults,
+            Map<String, String> governedUserToProxy,
+            Map<String, String> governedUserToProxyExtraSearch) {
+        List<ParticipantsLookupResultRow> resultList = new ArrayList<>();
+        for (var participant : participantsStructuredResults.values()) {
+            ParticipantsLookupResultRow resultRow = new ParticipantsLookupResultRow(participant);
+            var proxyGuid = governedUserToProxy.get(participant.getGuid());
+            if (proxyGuid != null) {
+                resultRow.setProxy(new ParticipantsLookupResultRowBase(usersResults.get(proxyGuid)));
+            } else if (participant.getProxies().size() > 0) {
+                governedUserToProxyExtraSearch.put(participant.getGuid(), participant.getProxies().get(0));
+            }
+            resultList.add(resultRow);
+        }
+        return resultList;
+    }
+
+    private List reMergeResultsWithExtraFoundProxies(
+            String query,
+            int resultsMaxCount,
+            Map<String, String> governedUserToProxy,
+            Map<String, String> governedUserToProxyExtraSearch,
+            String usersEsIndex,
+            Map proxiesResults,
+            Map participantsStructuredResults) throws IOException {
+        List esResultRows;
+        Map<String, ESUsersIndexResultRow> proxiesExtraResults = new ESUsersProxiesExtraSearch(esClient, governedUserToProxyExtraSearch)
+                .setResultMaxCount(resultsMaxCount)
+                .setEsIndex(usersEsIndex)
+                .setFetchSource(USERS__INDEX__SOURCE)
+                .setQuery(query)
+                .search();
+        // add to proxy result found extra proxies (which were not found during search step 1)
+        for (var proxyGuid : proxiesExtraResults.keySet()) {
+            proxiesResults.put(proxyGuid, proxiesExtraResults.get(proxyGuid));
+        }
+        // copy to map governedUser=proxy the found extra proxies
+        governedUserToProxy.putAll(governedUserToProxyExtraSearch);
+        // repeated merge of results
+        esResultRows = mergeResults(
```

Comment:
This will create twice the amount of objects... anyway to reuse `esResultRows` from the first merge and just fill in what's missing?